src/Sulu/Bundle/EventLogBundle/UserInterface/Controller/EventRecordController.php

```diff
@@ -0,0 +1,450 @@
+                if (($permission->getPermissions() & $viewPermission) === $viewPermission) {
+                    $securityContexts[] = $permission->getContext();
+                }
+            }
+        }
+
+        $securityContexts = \array_unique($securityContexts);
+
+        $listBuilder->addExpression(
+            $listBuilder->createOrExpression(
+                [
+                    $listBuilder->createInExpression(
+                        $fieldDescriptors['resourceSecurityContext'],
+                        $securityContexts
+                    ),
+                    $listBuilder->createWhereExpression(
+                        $fieldDescriptors['resourceSecurityContext'],
+                        null,
+                        ListBuilderInterface::WHERE_COMPARATOR_EQUAL
+                    ),
+                ]
+            )
+        );
+    }
+
+    /**
+     * @param array<string, FieldDescriptorInterface> $fieldDescriptors
+     */
+    private function addResourceLocaleCondition(
+        DoctrineListBuilder $listBuilder,
+        array $fieldDescriptors,
+        string $resourceLocale
+    ): void {
+        $listBuilder->addExpression(
+            $listBuilder->createOrExpression(
+                [
+                    $listBuilder->createWhereExpression(
+                        $fieldDescriptors['resourceLocale'],
+                        $resourceLocale,
+                        ListBuilderInterface::WHERE_COMPARATOR_EQUAL
+                    ),
+                    $listBuilder->createWhereExpression(
+                        $fieldDescriptors['resourceLocale'],
+                        null,
+                        ListBuilderInterface::WHERE_COMPARATOR_EQUAL
+                    ),
+                ]
+            )
+        );
+    }
+
+    /**
+     * @param array<string, FieldDescriptorInterface> $fieldDescriptors
+     */
+    private function addResourceKeyCondition(
+        DoctrineListBuilder $listBuilder,
+        array $fieldDescriptors,
+        string $resourceKey
+    ): void {
+        $listBuilder->where(
+            $fieldDescriptors['resourceKey'],
+            $resourceKey,
+            ListBuilderInterface::WHERE_COMPARATOR_EQUAL
+        );
+    }
+
+    /**
+     * @param array<string, FieldDescriptorInterface> $fieldDescriptors
+     */
+    private function addResourceIdCondition(
+        DoctrineListBuilder $listBuilder,
+        array $fieldDescriptors,
+        string $resourceId
+    ): void {
+        $listBuilder->where(
+            $fieldDescriptors['resourceId'],
+            $resourceId,
+            ListBuilderInterface::WHERE_COMPARATOR_EQUAL
+        );
+    }
+
+    /**
+     * @param array<string, mixed> $eventRecord
+     */
+    private function getEventText(array $eventRecord, string $translationLocale): string
+    {
+        $translationParameters = [];
+        foreach ($eventRecord as $key => $value) {
```

Comment:
think it would be nice to do the same for the context of the event. right now its not possible to access that data in the translation right?